runtime-modules/creator-token/crt-core/src/lib.rs

```diff
@@ -0,0 +1,694 @@
+use codec::FullCodec;
+use core::default::Default;
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage,
+    dispatch::{fmt::Debug, marker::Copy, DispatchError, DispatchResult},
+    ensure,
+};
+use sp_arithmetic::traits::{AtLeast32BitUnsigned, One, Saturating, Zero};
+
+mod types;
+use types::{AccountDataOf, MaxTotalIssuance, TokenDataOf};
+
+/// Pallet Configuration Trait
+pub trait Trait: frame_system::Trait {
+    /// Events
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+
+    // TODO: Add frame_support::pallet_prelude::TypeInfo trait
+    /// the Balance type used
+    type Balance: AtLeast32BitUnsigned + FullCodec + Copy + Default + Debug + Saturating;
+
+    /// The token identifier used
+    type TokenId: AtLeast32BitUnsigned + FullCodec + Copy + Default + Debug;
+}
+
+/// The Base Token Trait
+pub trait MultiCurrency<T: Trait> {
```

Comment:
Why do we need all of this as a trait? the interface to be used by the rest of the runtime is going to be minimal: issue + deissue + a few other things. These are things which either will be extrinsics (transfer, burn, ) or only used internally (stake). I think it would be better to at least start with a minimal API, and only add to it once we know we need it.

Reply:
Maybe what I had in mind (and poorly realized) is https://github.com/open-web3-stack/open-runtime-module-library/blob/master/traits/src/currency.rs
That is having several traits like `MultiCurrency` `MultiReservableCurrency` etc.. and then in `Creator Token` have a 
`type MultiCurrency = MultiCurrency + MultiReservableCurrency;` And then use the `project_token::Module` as the actual implementation